test/test_get.py

```diff
@@ -196,37 +210,147 @@ def test_download_zip_and_expiration():
+    assert cache["test_api"]["date_downloaded"]
+
     for path in paths:
         assert os.path.exists(path)
 
     # use files downloaded here and manipulate cache file to test expiration
     downloader.cache_dict["test_resource"]["date_downloaded"] = str(
         datetime.now() - timedelta(days=4)
     )
+    downloader.cache_dict["test_api"]["date_downloaded"] = str(
+        datetime.now() - timedelta(days=4)
+    )
 
-    paths = downloader.download(resource)
+    paths = downloader.download(resource, api)
     # should not download again
     assert "tmp" in paths[0]
+    assert "tmp" in paths[1]
 
     # minus 8 days from date_downloaded
     downloader.cache_dict["test_resource"]["date_downloaded"] = str(
         datetime.now() - timedelta(days=8)
     )
+    downloader.cache_dict["test_api"]["date_downloaded"] = str(
+        datetime.now() - timedelta(days=8)
+    )
 
-    paths = downloader.download(resource)
+    paths = downloader.download(resource, api)
     # should download again
     assert paths[0] is not None
+    assert paths[1] is not None
+
+
+@pytest.mark.parametrize(
+    "downloader",
+    [
+        "downloader_without_specified_cache_dir",
+        "downloader_with_specified_cache_dir",
+    ],
+    indirect=True,
+)
+def test_cache_api_request(downloader):
+    api1 = APIRequest(
+        name="uniprot_api",
+        url_s=[
+            "https://rest.uniprot.org/uniprotkb/P12345.json",
+            "https://rest.uniprot.org/uniprotkb/P69905.json",
+        ],
+        lifetime=1,
+    )
+    api2 = APIRequest(
+        name="intact_api",
+        url_s="https://www.ebi.ac.uk/intact/ws/interactor/countTotal",
+        lifetime=1,
+    )
+
+    paths = downloader.download(api1, api2)
+
+    # test download list
+    assert isinstance(paths, list)
+    assert len(paths) == 3  # 2 API requests from api1, 1 API request from api2
+    for path in paths:
+        assert os.path.exists(path)
+    assert f"{os.sep}{api1.name}{os.sep}P12345.json" in paths[0]
+    assert f"{os.sep}{api1.name}{os.sep}P69905.json" in paths[1]
+    assert f"{os.sep}{api2.name}{os.sep}countTotal.json" in paths[2]
+
+    # api1 and api2 have been cached
+    # test cached api request
+    test_paths = downloader.download(
+        api1, api2
+    )  # get the path(s) of cached API request(s)
+    assert isinstance(test_paths, list)
+    assert len(paths) == len(test_paths)
+
+    paths.sort()
+    test_paths.sort()
+    for i in range(len(paths)):
+        with open(paths[i], "r") as file1:
+            api_request1 = json.load(file1)
+
+        with open(test_paths[i], "r") as file2:
+            api_request2 = json.load(file2)
+        assert api_request1 == api_request2
+
+
+@pytest.mark.parametrize(
+    "downloader",
+    [
+        "downloader_without_specified_cache_dir",
+        "downloader_with_specified_cache_dir",
+    ],
+    indirect=True,
+)
+def test_path_splitted(downloader):
```

Comment:
What is the purpose of this test, what does "path splitted" mean?

Reply:
@slobentanzer When `download` method handles multiple resources, where one resource has a single file and another has multiple, the path(the path for resource1) will be split into string characters...so I modified the `_download_resource` to return a list.In the test, if the length pf paths is 3(one for resource1 and two from resource2), then path for resource1 is returned correctly.